core/src/main/java/com/linecorp/armeria/client/endpoint/AbstractDynamicEndpointGroupBuilder.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.endpoint;
+
+import com.linecorp.armeria.client.Endpoint;
+import com.linecorp.armeria.common.annotation.UnstableApi;
+
+/**
+ * A skeletal builder implementation for a {@link DynamicEndpointGroup} and its subtypes.
+ */
+@UnstableApi
+public abstract class AbstractDynamicEndpointGroupBuilder {
```

Comment:
note: DNS cache PR introduced `AbstractDnsResolverBuilder` that conflicts with this builder.
`AbstractDynamicEndpointGroupBuilder` should extend `AbstractDnsResolverBuilder`.
https://github.com/line/armeria/pull/4133/files#diff-25f820493abf12e84fab771a6ef1ab4a6537fade6b9d31b1e95172d74be3f3e9R38

Reply:
I was confused.
`DnsEndpointGroupBuilder` needs to extend both `AbstractDynamicEndpointGroupBuilder` and `AbstractDnsResolverBuilder`.
Let me handle it in #4133 once this PR is merged.